### PR TITLE
Added new auth response format

### PR DESCRIPTION
**Pull request by bporter125**
I am writing tests, and fixing failing ones, but this is functionally complete for auth denial context and the new auth response format. I wanted to give you guys a chance to give some feedback as I still have a decent amount more work left regarding unit tests.

### CODE REVIEW

File: tests/test_entities_service.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        self.assertEqual(response.reason, AuthResponseReason.OTHER)
+
+    @data(Invalid, TypeError, ValueError)
+    def test_authorization_response_response_jwe_unexpected_device_response(
+            self, exc):
+        self.transport.decrypt_response.side_effect = exc
+        with self.assertRaises(UnexpectedDeviceResponse):
+            AuthorizationResponse(self.data, self.transport)
+
+    def test_authorization_response_rsa_response_data(self):
```
Comment (aenglander) [inline tests/test_entities_service.py]: I'm also testing that the response has null for type, reason, and denial_reason. Not sure if it's necessary but I'm making sure they are null as expected.

File: tests/test_entities_service.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+            "type": "AUTHORIZED",
+            "reason": "APPROVED",
+            "denial_reason": "32",
+            "service_pins": ["1", "2", "3"],
+            "device_id": "31e5b804-f9a7-11e8-97ef-0242ac130008"
+        }
+
+        self.addCleanup(patch.stopall)
+
+    def test_authorization_response_jwe_response_data(self):
```
Comment (aenglander) [inline tests/test_entities_service.py]: There is no verification of denial_reason in the response.

File: launchkey/entities/service.py [hunk 3]
```diff
@@ -419,18 +421,14 @@ def _parse_device_response_from_auth_package(self, auth_package, key_id,
             data = transport.decrypt_rsa_response(
                 auth_package, key_id)
             unmarshalled_package = loads(data)
-            decrypted_package = AuthorizationResponsePackageValidator().\
+            decrypted_package = AuthorizationResponsePackageValidator.\
```
[the remark this answers is not in the document]
Comment (bporter125) [reply launchkey/entities/service.py]: That's how it was doing it before, and I thought I was fixing a bug. After instantiating it, it resulted in unit tests failing so I just reverted the change.

File: tests/test_entities_service.py (hunk 2 above)
Comment (bporter125) [reply tests/test_entities_service.py]: Good catch. I added that as well as the type and reason to make sure their raw values get passed through.

File: tests/test_entities_service.py (hunk 1 above)
Comment (bporter125) [reply tests/test_entities_service.py]: I added those as well as fraud.
